src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUncommentedMainTest.java

```diff
@@ -0,0 +1,66 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.UncommentedMainCheck;
+
+public class XpathRegressionUncommentedMainTest extends AbstractXpathTestSupport {
+
+    private final Class<UncommentedMainCheck> clazz =
+            UncommentedMainCheck.class;
+
+    @Override
+    protected String getCheckName() {
+        return clazz.getSimpleName();
+    }
+
+    @Test
+    public void testOne() throws Exception {
```

Comment:
@AmrDeveloper , please add one more test, we need at least two tests.

Reply:
@romani Done but CI not passed again !

Reply:
I don’t think the pitest failure is transient/flaky, but I tried looking at how to resolve it yesterday and didn’t find the answer...

A common approach could be to take the line of code and the mutation that is surviving and construct a test that fails when that mutation is present, but the confusing thing here is pitest lists two identical mutations for the same line with one surviving, so I’m not sure what the report is telling us.

Reply:
> but CI not passed again !

it is even better, at least behavior is stable. It is sad that it blocks your PR. Looks like we run into pitest bug, so we need to find a workaround....